eslint.config.js

```diff
@@ -0,0 +1,316 @@
+import jqueryConfig from "eslint-config-jquery";
+import importPlugin from "eslint-plugin-import";
+import globals from "globals";
+
+export default [
+	{
+
+		// Only global ignores will bypass the parser
+		// and avoid JS parsing errors
+		// See https://github.com/eslint/eslint/discussions/17412
+		ignores: [
+			"test/data/json_obj.js"
+		]
+	},
+
+	{
+		files: [
+			"eslint.config.js",
+			"Gruntfile.cjs",
+			"test/node_smoke_tests/commonjs/**",
+			"test/node_smoke_tests/module/**",
+			"test/promises_aplus_adapters/**",
+			"test/middleware-mockserver.cjs"
+		],
+		languageOptions: {
+			globals: {
+				...globals.node,
+				...globals.es2021
+			}
+		},
+		rules: {
+			...jqueryConfig.rules,
+			strict: [ "error", "global" ]
+		}
+	},
+
+	// Source
+	{
+		files: [ "src/**" ],
+		plugins: {
+			import: importPlugin
+		},
+		languageOptions: {
+			globals: {
+				window: false
+			}
+		},
+		rules: {
+			...jqueryConfig.rules,
+			"import/extensions": [ "error", "always" ],
+			"import/no-cycle": "error",
+
+			// "import/no-unused-modules": [
+			// 	"error",
+			// 	{
+			// 		unusedExports: true,
+
+			// 		// When run via WebStorm, the root path against which these paths
+			// 		// are resolved is the path where this ESLint config file lies,
+			// 		// i.e. `src`. When run via the command line, it's usually the root
+			// 		// folder of the jQuery repository. This pattern intends to catch both.
+			// 		// Note that we cannot specify two patterns here:
+			// 		//     [ "src/*.js", "*.js" ]
+			// 		// as they're analyzed individually and the rule crashes if a pattern
+			// 		// cannot be matched.
+			// 		ignoreExports: [ "{src/,}*.js" ]
+			// 	}
+			// ],
+			indent: [
+				"error",
+				"tab",
+				{
+					outerIIFEBody: 0
+				}
+			],
+			"one-var": [ "error", { var: "always" } ],
+			strict: [ "error", "function" ]
+		}
+	},
+
+	{
+		files: [ "src/wrapper.js" ],
+		languageOptions: {
+			sourceType: "script",
+			globals: {
+				jQuery: false,
+				module: true
+			}
+		},
+		rules: {
+			"no-unused-vars": "off",
+			indent: [
+				"error",
+				"tab",
+				{
+
+					// Unlike other codes, "wrapper.js" is implemented in UMD.
+					// So it required a specific exception for jQuery's UMD
+					// Code Style. This makes that indentation check is not
+					// performed for 1 depth of outer FunctionExpressions
+					ignoredNodes: [
+						"Program > ExpressionStatement > CallExpression > :last-child > *"
+					]
+				}
+			]
+		}
+	},
+
+	{
+		files: [ "src/wrapper-esm.js" ],
+		languageOptions: {
+			globals: {
+				jQuery: false
+			}
+		},
+		rules: {
+			"no-unused-vars": "off",
+			indent: [
+				"error",
+				"tab",
+				{
+
+					// Unlike other codes, "wrapper.js" is implemented in UMD.
```

Comment:
It's already on `main` like that but I noticed this comment is not correct; `wrapper-esm.js` is not UMD (and the wrong file is referenced here anyway). It's just a wrapper which necessitates different rules but it's definitely not UMD (`wrapper.js` is not really a true UMD either, BTW 😉).

Reply:
I've updated to a simple comment about the purpose of the line, so that it does not indent code within the wrapper.